src/oc_erchef/apps/oc_chef_wm/src/oc_chef_wm_base.erl

```diff
@@ -444,22 +444,30 @@ authorized_by_org_membership_check(Req, #base_state{requestor=#chef_requestor{ty
     {true, Req, State};
 authorized_by_org_membership_check(Req, #base_state{organization_name = undefined} = State) ->
     {true, Req, State};
-authorized_by_org_membership_check(Req, #base_state{organization_name = OrgName, chef_db_context = DbContext} = State) ->
+authorized_by_org_membership_check(Req, #base_state{organization_name = OrgName,
+                                                    chef_db_context = DbContext,
+                                                    requestor = #chef_requestor{authz_id = RequestorAuthzId}} = State) ->
     {UserName, BypassesChecks} = get_user(Req, State#base_state{superuser_bypasses_checks = true}),
     case BypassesChecks of
         true -> {true, Req, State};
         _ ->
-            case chef_db:is_user_in_org(DbContext, UserName, OrgName) of
-                true ->
-                    {true, Req, State};
-                false ->
-                    Msg = forbidden_message(not_member_of_org, UserName, OrgName),
-                    {false, wrq:set_resp_body(chef_json:encode(Msg), Req),
-                     State#base_state{log_msg = user_not_in_org}};
-                Error ->
-                    Msg = forbidden_message(unverified_org_membership, UserName, OrgName),
-                    {false, wrq:set_resp_body(chef_json:encode(Msg), Req),
-                     State#base_state{log_msg = {user_not_in_org_error, Error}}}
+            %% If user is a member of the global group server-admins,
+            %% we do not want to fail on org membership check, only acls.
+            case is_server_admin(DbContext, RequestorAuthzId) of
+                true -> {true, Req, State};
+                _ ->
+                    case chef_db:is_user_in_org(DbContext, UserName, OrgName) of
```

Comment:
I would guess for 90% of our requests the requestor is *not* a server admin, but is in the org. We should swap the two fetches and most likely save a request.